tree/ntuple/v7/doc/specifications.md

```diff
@@ -1057,27 +1065,51 @@ The limits refer to a single RNTuple and do not consider combinations/joins such
 
 ## Glossary
 
-TODO: This glossary is not yet complete
+### Anchor
+
+The anchor is a data block that makes the entry point to an RNTuple.
+The anchor is specific to the RNTuple container in which the RNTuple data are embedded (e.g., a ROOT file).
+The anchor must provide the information to load the header and the footer **envelopes**.
+
+### Cluster
+
+A cluster is a set of **pages** that contain all the data that belongs to a certain entry range.
+The data set is partitioned in clusters.
+A typical cluster size is tens to hundreds of megabytes.
 
 ### Column
 
 A column is a storage backed vector of a number of **elements** of a simple type.
-Column elements have a fixed bit-length depending on the column type.
+Column elements have a fixed bit-length that depends on the column type.
+Some column types allow setting the bit lengths within limits, (e.g. for floats with truncated mantiassa).
+
+### Envelope
+
+An envelope is a data block with RNTuple meta-data, such as the header and the footer.
+
+### Field
+
+A field describes a serialized C++ type.
+A field can have a hierarchy of subfields representing a composed C++ type (e.g., a vector of integers).
+A field has zero, one, or multiple **columns** attached to it.
+The columns contain the data related to the field but not to its subfields, which have their own columns.
+
+### Frame
+
+A frame is a byte range with meta-data information in an **envelope**.
```

Comment:
```suggestion
A frame is a byte range with metadata information in an **envelope**.
```
Other places in the text don't hyphenate. I don't have a specific preference, but we should be consistent.